corehq/apps/api/urls.py

```diff
@@ -1,3 +1,28 @@
+"""
+The CommCareHQ API is comprised of resources, representing individual data
+models. Each resource may contain multiple endpoints (list, detail, etc), and
+may have multiple versions.
+
+Historically, resources were versioned together, so making a backwards
+incompatible change to one required bumping the version for everything. This
+was our standard up through API version 0.6. URLS looked like this:
+
+    /a/mydomain/api/v0.5/user/
+
+In 2024 we switched to versioning resources independently and dropped the minor
+versioning, such that new URLs look like this, with the resource name first:
+
+    /a/mydomain/api/user/v1/
+
+To manage the transition, all v0.5 URls are duplicated as v1, and v0.6 URLs are
+duplicated as v2.
```

Comment:
(copying here from one of my commit messages)

One hesitation I have about this change is that it modifies the URL returned by `reverse` for each of the non tastypie views to the new latest version. That means for example, that those of these which return a 'next' parameter will have it now pointing to the newer version.  The view is the same in both instances, so the only way I can imagine it breaking is if the client has something that checks the format of the URL returned, but I don't like the behavior change.

This isn't the case where there's actually a separate view for each version, only when both versions use the same exact code and url name.  Actually this isn't a problem with tastypie resources, as the version number is one of the URL kwargs that must be passed in to reverse, allowing the URL resolver to disambiguate.  I could probably adopt a similar pattern elsewhere if this is a concern, it just means changing each of the views individually.